libp2p/stream/connection.nim

```diff
@@ -53,7 +56,8 @@ proc onUpgrade*(s: Connection) {.async.} =
   if not isNil(s.upgraded):
     await s.upgraded
 
-func shortLog*(conn: Connection): string =
+func shortLog*(conn: Connection): string
+  {.raises: [Defect, ValueError].} =
```

Comment:
`shortLog` should never raise - in fact it doesn't _actually_ except that `&` raises - the right thing to do here is to catch the `ValueError` and turn it into a `Defect`

Reply:
Yeah, quick and dirty fix to get things to compile on first pass, but I suspected that this wasn't the right approach.